src/ImageSharp/Formats/Tga/TgaImageFormatDetector.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+
+namespace SixLabors.ImageSharp.Formats.Tga
+{
+    /// <summary>
+    /// Detects tga file headers.
+    /// </summary>
+    public sealed class TgaImageFormatDetector : IImageFormatDetector
+    {
+        /// <inheritdoc/>
+        public int HeaderSize => TgaConstants.FileHeaderLength;
+
+        /// <inheritdoc/>
+        public IImageFormat DetectFormat(ReadOnlySpan<byte> header)
+        {
+            return this.IsSupportedFileFormat(header) ? TgaFormat.Instance : null;
+        }
+
+        private bool IsSupportedFileFormat(ReadOnlySpan<byte> header)
+        {
+            if (header.Length >= this.HeaderSize)
+            {
+                // There is no magick bytes in a tga file, so at least the image type
+                // and the colormap type in the header will be checked for a valid value.
+                if (header[1] != 0 && header[1] != 1)
```

Comment:
Astonishing that there are no identifiers!

Reply:
yes its really strange. As far as i know its the only image format without some sort of magick bytes